doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,317 @@
+.. _soc_porting_guide:
+
+SoC Porting Guide
+###################
+
+To add Zephyr support for a new :term:`SoC`, you at least need a *SoC directory*
+with various files in it. Also a SoC :file:`.dtsi` in
+:zephyr_file:`dts/<ARCH>/<VENDOR>` is required.
+
+SoC Definitions
+***************
+
+It is expected that you are familiar with the board concept in Zephyr.
+A high level overview of the hardware support hierarchy and terms used in the
+Zephyr documentation can be seen in :ref:`hw_support_hierarchy`.
+
+For SoC porting, the most important terms are:
+
+- SoC: the exact system on a chip the board's CPU is part of
+- SoC series: a smaller group of tightly related SoCs
+- SoC family: a wider group of SoCs with similar characteristics
+- CPU Cluster: a cluster of one or more CPU cores.
+- CPU core: a particular CPU in an architecture
+- Architecture: an instruction set architecture
+
+Architecture
+============
+
+See :ref:`architecture_porting_guide`.
+
+
+Create your SoC directory
+*************************
+
+Each SoC must have a unique name. You need to give your SoC a unique name.
+The script ``list_hardware`` can be used to retrieve a list of all SoCs known
+in Zephyr, for example ``./scripts/list_hardware --soc-root=. --socs`` from
+Zephyr base for a list of names that are already taken, and pick something new.
```

Comment:
`and pick something new.` This leaves place to arbitrary naming which we want to avoid:
`and based on the existing naming scheme, add the reference that matches your SoC`

Reply:
Rephrased based on feedback from @nordicjm